icu4c/source/test/cintltst/cbiapts.c

```diff
@@ -515,6 +517,79 @@ static void TestBreakIteratorSafeClone(void)
         ubrk_close(someIterators[i]);
     }
 }
+
+static void TestBreakIteratorClone(void)
+{
+    UChar text[51];     /* Keep this odd to test for 64-bit memory alignment */
+                        /*  NOTE:  This doesn't reliably force mis-alignment of following items. */
+    UBreakIterator * someIterators [CLONETEST_ITERATOR_COUNT];
+    UBreakIterator * someClonedIterators [CLONETEST_ITERATOR_COUNT];
+
+    UErrorCode status = U_ZERO_ERROR;
+    int32_t start,pos;
+    int32_t i;
+
+    /*Testing ubrk_clone */
+
+    /* Note:  the adjacent "" are concatenating strings, not adding a \" to the
+       string, which is probably what whoever wrote this intended.  Don't fix,
+       because it would throw off the hard coded break positions in the following
+       tests. */
+    u_uastrcpy(text, "He's from Africa. ""Mr. Livingston, I presume?"" Yeah");
+
+    /* US & Thai - rule-based & dictionary based */
```

Comment:
It is unclear from the code that CLONETEST_ITERATOR_COUNT is 2. If somehow that value is 3 then the following code in the loop won't work. Could we put the ubrk_open call also into the for loop to make it clear?

Reply:
I assumed that this test function is a modified copy of the safeClone test, and so personally I am inclined to be lenient here. It's true that this test code is not super. What do you think?

Reply:
I would be inclined to clean this up, as long as it's being touched anyhow. Keeping only the most basic check for the function's existence in cintltst, and having all else in the testing of BreakIterator::clone() in intltest. But I don't feel all that strongly about it.